File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/user/SendForgotPasswordCodeUseCase.java
package com.lattis.ellipse.domain.interactor.user;

import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.UserRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 5/2/17.
 */

public class SendForgotPasswordCodeUseCase extends UseCase<BasicResponse> {

    private UserRepository userRepository;
    private String email;

    @Inject
    protected SendForgotPasswordCodeUseCase(ThreadExecutor threadExecutor,
                                          PostExecutionThread postExecutionThread,
                                          UserRepository userRepository) {
        super(threadExecutor, postExecutionThread);
        this.userRepository = userRepository;
    }

    public SendForgotPasswordCodeUseCase toEmail(String email){
        this.email=email;
        return this;
    }

    @Override
    protected Observable<BasicResponse> buildUseCaseObservable() {
        return userRepository.sendForgotPasswordCode(email);
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/user/SendForgotPasswordCodeUseCase.kt
package com.lattis.domain.usecase.user

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.UserRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class SendForgotPasswordCodeUseCase @Inject protected constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val userRepository: UserRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var email: String? = null
    fun toEmail(email: String?): SendForgotPasswordCodeUseCase {
        this.email = email
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return userRepository.sendForgotPasswordCode(email!!)
    }

}